apps/fem/poisson3.sim

```diff
@@ -0,0 +1,188 @@
+%
+%  Strong form (poisson problem)
+%
+% - \nabla^2 u = f(x,y) in \Omega (domain)
+%            u = 0      on \Gamma (boundary)
+%
+%  solved using linear functions on a tetrahedral mesh
+%
+
+element Node
+    x    : vector[3](float); % position
+    edge : bool;             % true if part of the boundary
+    u    : float;            % solution
+    i    : int;              % index of node
```

Comment:
Already, I'm assuming that `Set`is keeping its indexing somewhat sensible.  That is, I have a nested for loop putting things into the `Set` and later iterate over this to write results by typing `for(auto el : mySimitSet){ ... }`. If the latter returns the elements in any other order than the one I put them in, then my write function will create bogus output.

Is this a reasonable assumption? I'm happy with the current solution as long as my Set iterator behaves "naturally".

If you reorganize the indexing behind the scenes and files being read, results in different order when writing, I expect there will be a lot of confusion and errors. Also cumbersome to work with. For me, when writing the nodal points, I would have to do a linear search for each point, looking something like this
```c++
simit::Set nodes;

FieldRef<int> i = node.getField<int>("i");
FieldRef<double,3> x = node.getField<double,3>("x");
int n = nodes.getSize();

for(int j=0; j<n; j++) {
  for(auto e : nodes) {
    if(i(e) == j)
      std::cout << x(e) << std::endl;
  }
}
```